johannesjo/super-productivity
# Add Ukrainian (uk) language support

# Summary

This pull request adds support for the Ukrainian language (uk) to the application. It includes adding the language to the list of supported languages, updating various language-related configurations, and adding the Ukrainian translation file.

# Files Changed

1. `project.inlang/settings.json`: Added "uk" to the `languageTags` array.
2. `src/app/app.constants.ts`: Added Ukrainian language support to various language-related enums and imports.
3. `src/app/features/config/form-cfgs/language-selection-form.const.ts`: Added Ukrainian language option to the language selection form.
4. `src/app/t.const.ts`: Added Ukrainian language constant.
5. `src/assets/i18n/en.json`: Added Ukrainian language name to the English translations.
6. `src/assets/i18n/uk.json`: Added new file containing Ukrainian translations.

# Code Changes

## `project.inlang/settings.json`

```json
"languageTags": [
  // ...
  "tr",
  "uk",  // Added Ukrainian language code
  "zh",
  // ...
]
```

## `src/app/app.constants.ts`

```typescript
import '@angular/common/locales/global/uk';  // Added import for Ukrainian locale

export enum LanguageCode {
  // ...
  uk = 'uk',  // Added Ukrainian language code
  // ...
}

export enum LanguageCodeMomentMap {
  // ...
  uk = 'uk',  // Added Ukrainian language code mapping
  // ...
}
```

## `src/app/features/config/form-cfgs/language-selection-form.const.ts`

```typescript
{ label: T.GCF.LANG.UK, value: LanguageCode.uk },  // Added Ukrainian language option
```

## `src/app/t.const.ts`

```typescript
UK: 'GCF.LANG.UK',  // Added Ukrainian language constant
```

## `src/assets/i18n/en.json`

```json
"UK": "Українська",  // Added Ukrainian language name in English
```

## `src/assets/i18n/uk.json`

A new file was added containing the complete Ukrainian translation for the application. This file includes translations for all UI elements, messages, and settings.

# Reason for Changes

These changes were made to add support for the Ukrainian language in the application. This will allow Ukrainian-speaking users to use the application in their native language, improving accessibility and user experience for this demographic.

# Impact of Changes

1. Ukrainian-speaking users will now be able to select Ukrainian as their preferred language in the application.
2. All UI elements, messages, and settings will be displayed in Ukrainian when the language is selected.
3. The application's language selection options will now include Ukrainian.
4. The change is backwards compatible and does not affect existing language selections or functionality.

# Test Plan

1. Verify that Ukrainian appears as an option in the language selection settings.
2. Select Ukrainian as the application language and confirm that all UI elements, messages, and settings are correctly translated.
3. Ensure that switching between Ukrainian and other languages works as expected.
4. Test the application's functionality with Ukrainian selected to ensure no issues arise from the new translations.
